src/librustc_mir/transform/const_prop.rs

```diff
@@ -692,6 +587,18 @@ impl<'mir, 'tcx> MutVisitor<'tcx> for ConstPropagator<'mir, 'tcx> {
                     }
                 }
             }
+        } else if let StatementKind::StorageLive(local) = statement.kind {
+            if self.can_const_prop[local] {
+                let frame = self.ecx.frame_mut();
+
+                frame.locals[local].value = LocalValue::Uninitialized;
+            }
+        } else if let StatementKind::StorageDead(local) = statement.kind {
+            if self.can_const_prop[local] {
+                let frame = self.ecx.frame_mut();
+
+                frame.locals[local].value = LocalValue::Dead;
```

Comment:
This `if` branch is repeated here from the other `else if let` branch; only diff is the variant in `.value =LocalValue::Variant;`.